Error tagging job version when trying to tag any job

### Nomad version
1.9.7

### Operating system and Environment details
Clients & Servers: Debian 12
User OS: Ubuntu 24.04
User Browser. Firefox 139.0.1

### Issue
When trying to tag any job in the Nomad Web UI, it fails with an error pop up like
```
Error Tagging Job Version
job "nomad-debug-job" version 0 not found
```
On the server I see the same error, shown further below.

With the CLI , I can tag job versions without issue, as long as the `-namespace` parameter is provided. However, without the `-namespace` parameter, I get the same error. This leads me to believe that the Web UI does not correctly supply the `namespace` parameter when calling the Nomad API.

### Reproduction steps
Try to tag any job version in the nomad Web UI.

#### Expected Result
The tag is created.

#### Actual Result
The above mentioned error occurs.

### Nomad Server logs (if appropriate)
2025-07-02T07:36:18.194Z [ERROR] nomad.fsm: UpdateJobVersionTag failed: error="job \"nomad-debug-job\" version 0 not found"

--- a/.changelog/27282.txt
+++ b/.changelog/27282.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+ui: Tagging job versions in another namespace than the default-namespace resulted in an error
+```

--- a/ui/app/adapters/version-tag.js
+++ b/ui/app/adapters/version-tag.js
@@ -13,18 +13,19 @@ export default class VersionTagAdapter extends ApplicationAdapter {
   urlForCreateRecord(_modelName, model) {
     const tagName = model.attr('name');
     const jobName = model.attr('jobName');
-    return `${this.buildURL()}/job/${jobName}/versions/${tagName}/tag`;
+    const namespace = model.attr('jobNamespace');
+    return `${this.buildURL()}/job/${jobName}/versions/${tagName}/tag?namespace=${namespace}`;
   }
 
-  async deleteTag(jobName, tagName) {
+  async deleteTag(namespace, jobName, tagName) {
     let deletion = this.ajax(
-      this.urlForDeleteRecord(jobName, tagName),
+      this.urlForDeleteRecord(namespace, jobName, tagName),
       'DELETE'
     );
     return deletion;
   }
 
-  urlForDeleteRecord(jobName, tagName) {
-    return `${this.buildURL()}/job/${jobName}/versions/${tagName}/tag`;
+  urlForDeleteRecord(namespace, jobName, tagName) {
+    return `${this.buildURL()}/job/${jobName}/versions/${tagName}/tag?namespace=${namespace}`;
   }
 }

--- a/ui/app/components/job-version.js
+++ b/ui/app/components/job-version.js
@@ -51,6 +51,7 @@ export default class JobVersion extends Component {
       this.editableTag = this.store.createRecord('versionTag');
     }
     this.editableTag.versionNumber = this.version.number;
+    this.editableTag.jobNamespace = this.version.get('job.namespace.name');
     this.editableTag.jobName = this.version.get('job.plainId');
   }
 
@@ -224,7 +225,11 @@ export default class JobVersion extends Component {
     try {
       await this.store
         .adapterFor('version-tag')
-        .deleteTag(this.editableTag.jobName, this.editableTag.name);
+        .deleteTag(
+          this.editableTag.jobNamespace,
+          this.editableTag.jobName,
+          this.editableTag.name
+        );
       this.notifications.add({
         title: 'Job Version Un-Tagged',
         color: 'success',

--- a/ui/app/models/version-tag.js
+++ b/ui/app/models/version-tag.js
@@ -11,5 +11,6 @@ export default class VersionTagModel extends Fragment {
   @attr() description;
   @attr() taggedTime;
   @attr('number') versionNumber;
+  @attr('string') jobNamespace;
   @attr('string') jobName;
 }

